forwarder.go

```diff
@@ -226,28 +227,76 @@ func buildExternalURL(newTalariaName, domain string) string {
  * Extracts IP address, certificate provider, and expiry details from HTTP request headers.
  * Calls the RI API with this information.
  */
-func updateResourceIpAddressAndCertificateInfo(req *http.Request, client *http.Client, resourceURL *url.URL) error {
 
-	certificateProviderRaw := req.Header.Get(certificateProviderHeader)
+// base64Decode decodes a base64 encoded string.
+func base64Decode(encodedStr string) ([]byte, error) {
+	decodedData, err := base64.StdEncoding.DecodeString(encodedStr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode base64 string: %v", err)
+	}
+	return decodedData, nil
+}
+
+// extractWebPAConveyHeader extracts the X-WebPA-Convey header value.
+func extractWebPAConveyHeader(req *http.Request) (string, error) {
+	xWebPACode := req.Header.Get("X-WebPA-Convey")
+	if xWebPACode == "" {
+		return "", fmt.Errorf("X-WebPA-Convey header not found")
+	}
+	return xWebPACode, nil
+}
+
+// updateResourceIpAddressAndCertificateInfo updates resource information.
+func updateResourceIpAddressAndCertificateInfo(req *http.Request, client *http.Client, resourceURL *url.URL) error {
+	certificateProviderRaw := req.Header.Get("X-Certificate-Provider")
```

Comment:
What is the reason for changes the request header key for certificateProviderRaw?